custom/enikshay/data_store.py

```diff
@@ -32,10 +32,13 @@ def latest_adherence_date(self, episode_id):
         if result:
             # the result is sorted on 'adherence_date'
             latest_date = result[0].get('adherence_date')
-            if not latest_date or not parse_date(latest_date):
-                raise EnikshayTaskException("Adherence row {} does not or has invalid 'adherence_date'".format(
-                    result[0]))
-            else:
-                return parse_date(latest_date)
+            try:
+                return datetime.strptime(latest_date, '%Y-%m-%d').date()
+            except ValueError:
+                try:
+                    return datetime.strptime(latest_date, '%Y-%m-%dT%H:%M:%S').date()
```

Comment:
i think this is still going to have an issue if the string has timezone info on it. what i was saying before was that it's ok to use `parse()`, we should just first make sure that the string matches a regex like `^\d{4}-\d{2}-\d{2}.+$` first